File: src/PhotoVote/Event/Candidate/CandidateAdded.py
from typing import Optional

from PhotoVote.Event import Event


class CandidateAdded(Event):
    candidate_id: Optional[str] = None
    competition_id: Optional[str] = None
    election_id: Optional[str] = None
    name: Optional[str] = None
    description: Optional[str] = None
    image_url: Optional[str] = None
    image_caption: Optional[str] = None

File: src/client/src/PhotoVote/Events/Candidate/CandidateAdded.ts
import {Event} from '../Event'
import {randomUUID} from "node:crypto";

class CandidateAdded extends Event {
    candidateId?: string
    competitionId?: string
    electionId?: string
    name?: string
    description?: string
    imageUrl?: string
    imageCaption?: string

    constructor() {
        super(randomUUID());
    }
}

export { CandidateAdded }